qa/python_models/bls/model.py

```diff
@@ -248,6 +248,58 @@ def test_zero_length_io(self):
         output0 = pb_utils.get_output_tensor_by_name(infer_response, 'OUTPUT0')
         self.assertTrue(np.all(output0 == input0))
 
+    def test_bls_tensor_lifecycle(self):
+        model_name = 'dlpack_identity'
+
+        # A 10 MB tensor.
+        input_size = 10 * 1024 * 1024
+
+        # Sending the tensor 50 times to test whether the deallocation is
+        # happening correctly. If the deallocation doesn't happen correctly,
+        # there will be an out of shared memory error.
+        for _ in range(50):
+            input0 = np.ones([1, input_size], dtype=np.float32)
+            input0_pb = pb_utils.Tensor('INPUT0', input0)
+            infer_request = pb_utils.InferenceRequest(
+                model_name=model_name,
+                inputs=[input0_pb],
+                requested_output_names=['OUTPUT0'])
+            infer_response = infer_request.exec()
+            self.assertFalse(infer_response.has_error())
+
+            output0 = pb_utils.get_output_tensor_by_name(
+                infer_response, 'OUTPUT0')
+            self.assertTrue(np.all(output0.as_numpy() == input0))
```

Comment:
There's probably a better assertion for this. Maybe [numpy.testing.assert_equal](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_equal.html) for a clearer failure message?

Reply:
updated